### PR TITLE
Clarifying github example docs - http/https

### PR DESCRIPTION
**Pull request by charlesreid1**
- clarifying the purpose of OAUTHLIB_INSECURE_TRANSPORT
- "production" is less clear, use "local" or "trusted connection"
- add a few additional hints to the user about what's going on

### CODE REVIEW

File: docs/quickstarts/github.rst [hunk 1]
```diff
@@ -42,25 +42,28 @@ Code
     If you are running this code on Heroku, you'll need to use the
     :class:`werkzeug.contrib.fixers.ProxyFix` middleware. See :doc:`../proxies`.
 
-When you run this code, you must set the :envvar:`OAUTHLIB_INSECURE_TRANSPORT`
-environment variable for it to work. For example, if you put this code in a
-file named ``github.py``, you could run:
+If you run this code locally or without HTTPS enabled (see warning below), 
+you must set the :envvar:`OAUTHLIB_INSECURE_TRANSPORT` environment variable
+to disable the HTTPS requirement imposed by ``oauthlib``. For example, if
```
Comment (singingwolfboy) [inline docs/quickstarts/github.rst]: I expect that most people won't know what `oauthlib` is, and won't care. Can you change this to:

```
to disable the HTTPS requirement imposed by ``oauthlib``, which is part of Flask-Dance.
```

**Comment by singingwolfboy** [comment]
Hi there @charlesreid1, I see that this pull request is still in progress. Anything I can do to help you finish it up?

**Comment by charlesreid1** [comment]
Thanks for the ping. Have been busy with travel. But this hasn't dropped off my radar!

**Comment by singingwolfboy** [comment]
Great, enjoy your travel! Let me know if you have any idea when you can finish this up -- that way, I'll know not to ping you until then. Otherwise, I'll ping you every few weeks. :)

**Comment by charlesreid1** [comment]
All done and ready to merge

**Comment by singingwolfboy** [comment]
Thanks!

**Comment by singingwolfboy** [comment]
Oops, I just realized I forgot to merge this until now...

**Pull request merged**